app/Http/Controllers/FacebookController.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Interns2025b\Http\Controllers;
+
+use Illuminate\Http\JsonResponse;
+use Illuminate\Support\Facades\Auth;
+use Interns2025b\Models\User;
+use Laravel\Socialite\Facades\Socialite;
+use Symfony\Component\HttpFoundation\Response;
+
+class FacebookController extends Controller
+{
+    public function redirect(): JsonResponse
+    {
+        $url = Socialite::driver("facebook")->redirect()->getTargetUrl();
+
+        return response()->json(["url" => $url]);
+    }
+
+    public function loginCallback(): JsonResponse
+    {
+        $facebookUser = Socialite::driver("facebook")->user();
+        $email = $facebookUser->getEmail();
+
+        $existingUser = User::where("email", $email)->first();
+
+        if ($existingUser && $existingUser->facebook_id === null) {
+            return response()->json([
+                "message" => __("auth.email_already_registered"),
+            ], Response::HTTP_FORBIDDEN);
+        }
+
+        $nameParts = explode(" ", $facebookUser->getName(), 2);
+        $firstName = $nameParts[0] ?? "Facebook";
+
+        $user = User::firstOrCreate(
```

Comment:
Consider edge cases in name parsing - for example some users has two names :eyes: